test/e2e/parallel/spaceprovisionerconfig_test.go

```diff
@@ -110,4 +102,73 @@ func TestSpaceProvisionerConfig(t *testing.T) {
 			WithNameThat(spc.Name, Is(NotReady()))
 		require.NoError(t, err)
 	})
+	t.Run("becomes not ready when cluster becomes not ready", func(t *testing.T) {
+		// NOTE: this is impossible to test currently because there is no way for a TC to
+		// transition from ready to unready (short of updating the TC and restarting the toochaincluster cache controller
+		// which is something we can't afford in the parallel tests).
+		//
+		// // given
+		//
+		// // we need to create a copy of the cluster and the token secret
+		// existingCluster, err := host.WaitForToolchainCluster(t)
+		// require.NoError(t, err)
+		// cluster := copyClusterWithSecret(t, host.Awaitility, existingCluster)
+		//
+		// // when
+		// spc := CreateSpaceProvisionerConfig(t, host.Awaitility, ReferencingToolchainCluster(cluster.Name))
+		//
+		// // then
+		// _, err = wait.
+		// 	For(t, host.Awaitility, &toolchainv1alpha1.SpaceProvisionerConfig{}).
+		// 	WithNameThat(spc.Name, Is(Ready()))
+		// require.NoError(t, err)
+		//
+		// // when
+		// // update the TC such that it is no longer valid.
+		// require.NoError(t, host.Client.Get(context.TODO(), client.ObjectKeyFromObject(cluster), cluster))
+		// apiEndpoint, err := url.Parse(cluster.Spec.APIEndpoint)
+		// require.NoError(t, err)
+		// apiEndpoint.Host = apiEndpoint.Hostname() + "-not:" + apiEndpoint.Port()
+		// cluster.Spec.APIEndpoint = apiEndpoint.String()
+		// require.NoError(t, host.Client.Update(context.TODO(), cluster))
+		//
+		// // then
+		// _, err = wait.
+		// 	For(t, host.Awaitility, &toolchainv1alpha1.SpaceProvisionerConfig{}).
+		// 	WithNameThat(spc.Name, Is(NotReady()))
+		// require.NoError(t, err)
+	})
+}
+
+func copyClusterWithSecret(t *testing.T, a *wait.Awaitility, cluster *toolchainv1alpha1.ToolchainCluster) *toolchainv1alpha1.ToolchainCluster {
+	clusterName := util.NewObjectNamePrefix(t) + string(uuid.NewUUID()[0:20])
+	return copyClusterWithSecretAndNewName(t, a, clusterName, cluster)
+}
+
+func copyClusterWithSecretAndNewName(t *testing.T, a *wait.Awaitility, newName string, cluster *toolchainv1alpha1.ToolchainCluster) *toolchainv1alpha1.ToolchainCluster {
+	// copy the secret
+	secret := &corev1.Secret{}
+	require.NoError(t, a.CopyWithCleanup(t,
+		client.ObjectKey{
+			Name:      cluster.Spec.SecretRef.Name,
+			Namespace: cluster.Namespace,
+		},
+		client.ObjectKey{
+			Name:      newName,
+			Namespace: cluster.Namespace,
+		},
+		secret,
+	))
+
+	// and create a new ToolchainCluster with that secret
+	// note that we can't use the CopyWithCleanup function because we also
+	// need to modify the SecretRef of the TC.
+	newCluster := cluster.DeepCopy()
+	newCluster.ResourceVersion = ""
+	newCluster.UID = ""
+	newCluster.Name = newName
+	newCluster.Spec.SecretRef.Name = secret.Name
+	require.NoError(t, a.CreateWithCleanup(t, newCluster))
+
+	return newCluster
```

Comment:
this looks very similar to what we have in test/e2e/toolchaincluster_test.go file - do you think that we could unify the code in the testsupport/ package (or somewhere there)?

Reply:
Good point. Most of this could be changed to use `wait.CopyWithCleanup()`...